galley/pkg/config/analysis/analyzers/util/find_errorline_utils.go

```diff
@@ -0,0 +1,169 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	"istio.io/istio/pkg/config/resource"
+)
+
+// ErrorLineForHostInDestination returns the line number of the host in destination
+func ErrorLineForHostInDestination(routeRule string, serviceIndex int, destinationIndex int, r *resource.Instance) int {
```

Comment:
I have a few comments on the error functions. I'm going to make one comment, but these apply to all of the functions.

1) All of the other functions take the *resource.Instance as the first argument, can we change these to be the same?
2) I think that these should return (int, ok bool) so that if there's no error line found there's a way to check. In general using a zero value as "not found" isn't good practice, see https://golang.org/doc/effective_go.html#multiple-returns
3) Just return FindErrorLine(...) intead of setting a variable every time.